nodewatcher/modules/monitor/http/survey/management/commands/export_survey_data.py

```diff
@@ -0,0 +1,125 @@
+            type=int,
+            action='store',
+            dest='timestamp',
+            help="All the data will be collected between this UNIX timestamp and two hours preceding it."
+        )
+
+        parser.add_argument(
+            '-f',
+            '--filename',
+            type=str,
+            action='store',
+            dest='filename',
+            default='survey_export',
+            help="Optional filename",
+        )
+
+        parser.add_argument(
+            '--include-timestamp-in-filename',
+            type=bool,
+            action='store',
+            dest='store_timestamp',
+            default=False,
+            help="Include timestamp in filename?",
+        )
+
+    def handle(self, *args, **options):
+        """
+        Exports the latest survey data graph as a JSON file into the root directory.
+        """
+
+        timestamp = options['timestamp']
+        filename = options['filename']
+
+        exported_graph = extract_survey_graph(timestamp)
+
+        if not exported_graph:
+            self.stdout.write(self.style.ERROR("Insufficient survey data in the datastream during this time period."))
+            return
+
+        if options['store_timestamp']:
+            filename = '{0}{1}.json'.format(filename, timestamp)
+        else:
+            filename = '{0}.json'.format(filename)
+
+        with io.open(filename, 'w', encoding='utf-8') as f:
+            f.write(json.dumps(exported_graph, ensure_ascii=False))
+
+        self.stdout.write(self.style.SUCCESS("Successfully exported survey data."))
+
+
+def extract_survey_graph(timestamp=None):
+    """
+    Extracts a graph of the survey during a specified time period. Only the latest data
+    from specified datetime and two hours preceding the datetime will be used.
+
+    :param timestamp: UNIX timestamp according to the UTC timezone.
+    :return: A dictionary that contains the graph under the 'graph' key.
+    """
+
+    if timestamp:
+        latest_survey_datetime = datetime.datetime.fromtimestamp(timestamp)
+    else:
+        latest_survey_datetime = datetime.datetime.utcnow()
+
+    streams = datastream.find_streams({'module': 'monitor.http.survey'})
+
+    # Create a meta graph
+    meta_vertices, meta_edges = [], []
+    friendly_nodes = []
+    latest_data_timepoint = 0
+
+    for stream in streams:
+        datapoint_iterator = datastream.get_data(
+            stream_id=stream['stream_id'],
+            granularity=stream['highest_granularity'],
+            start=(latest_survey_datetime - datetime.timedelta(hours=2)),
+            reverse=True,
+        )
+        try:
+            stream_graph = datapoint_iterator[0]['v']
+            for vertex in stream_graph['v']:
+                meta_vertices.append(vertex)
+                if 'b' in vertex:
+                    friendly_nodes.append(vertex['i'])
+                    for bssid in vertex['b']:
+                        friendly_nodes.append(bssid)
```

Comment:
How should we construct this friendly_nodes list? This structure is most convenient for my algorithm but maybe not very legible.